src/shared/Core/GenericOAuthConfig.cs

```diff
@@ -1,32 +1,56 @@
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using GitCredentialManager.Authentication.OAuth;
 
 namespace GitCredentialManager
 {
     public class GenericOAuthConfig
     {
-        public static bool TryGet(ITrace trace, ISettings settings, Uri remoteUri, out GenericOAuthConfig config)
+        public static bool TryGet(ITrace trace, ISettings settings, InputArguments input, out GenericOAuthConfig config)
         {
             config = new GenericOAuthConfig();
+            Uri authzEndpointUri = null;
+            Uri tokenEndpointUri = null;
+            var remoteUri = input.GetRemoteUri();
 
-            if (!settings.TryGetSetting(
+            if (input.WwwAuth.Any(x => x.Contains("Basic realm=\"Gitea\"")))
+            {
+                trace.WriteLine($"Using universal Gitea OAuth configuration");
+                // https://docs.gitea.com/next/development/oauth2-provider?_highlight=oauth#pre-configured-applications
+                config.ClientId = "e90ee53c-94e2-48ac-9358-a874fb9e0662";
+                // https://docs.gitea.com/next/development/oauth2-provider?_highlight=oauth#endpoints
+                authzEndpointUri = new Uri(remoteUri, "/login/oauth/authorize");
+                tokenEndpointUri = new Uri(remoteUri, "/login/oauth/access_token");
+                config.RedirectUri = new Uri("http://127.0.0.1");
```

Comment:
Could we move some of these string literals to constants, perhaps a static inner-class `WellKnownConfig` or something? I can imagine this set of config growing in the future.

```suggestion
                config.ClientId = WellKnown.GiteaClientId;
                authzEndpointUri = new Uri(remoteUri, WellKnown.GiteaAuthzEndpoint);
                tokenEndpointUri = new Uri(remoteUri, WellKnown.GiteaTokenEndpoint);
                config.RedirectUri = WellKnown.LocalIPv4RedirectUri;
```

```csharp
public static class GenericOAuthConfig
{
    ...

    public static class WellKnown
    {
        // https://docs.gitea.com/next/development/oauth2-provider?_highlight=oauth#pre-configured-applications
        public const string GiteaClientId = "e90ee53c-94e2-48ac-9358-a874fb9e0662";
        // https://docs.gitea.com/next/development/oauth2-provider?_highlight=oauth#endpoints
        public const string GiteaAuthzEndpoint = "/login/oauth/authorize";
        public const string GiteaTokenEndpoint = "/login/oauth/access_token";
        public static Uri LocalIPv4RedirectUri = new Uri("http://127.0.0.1");
    }
}
```

Can also reference these magic strings in the unit tests too.

`GenericOAuthConfig.WellKnown.*`